src/main/java/HistoryServlet.java

```diff
@@ -0,0 +1,68 @@
+import com.google.gson.JsonObject;
+import com.google.gson.Gson;
+import com.google.sps.data.HistoryElement;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.lang.String;
+
+@WebServlet("/history")
+public class HistoryServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Put the current query from the user in datastore
+    String personType = request.getParameter("person-type");
+    String action = request.getParameter("action");
+    String location = request.getParameter("location");
+    String year = request.getParameter("year");
+    String userId = request.getParameter("user-id");
+
+    Entity historyEntity = new Entity("historyEntity");
+    historyEntity.setProperty("personType", personType);
+    historyEntity.setProperty("action", action);
+    historyEntity.setProperty("location", location);
+    historyEntity.setProperty("year", year);
+    historyEntity.setProperty("userId", userId);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(historyEntity);
+
+    HistoryElement requestHistoryElement = new HistoryElement(userId, 
+      personType, action, location, year);
+
+    Query query = new Query("historyEntity");
```

Comment:
You could try adding a [filter](https://cloud.google.com/appengine/docs/standard/java/datastore/queries#filters) to this query to only get the history for a particular userId so you don't have to check the userId in the for loop later

Reply:
I tried doing this earlier but I couldn't get it to compile because I wasn't constructing the filter properly. The code in the documentation works!